newbank/server/Customer.java

```diff
@@ -19,6 +22,14 @@ public String accountsToString() {
 	}
 
 	public void addAccount(Account account) {
-		accounts.add(account);		
+		accounts.add(account);
+	}
+
+	public boolean hasAccount(String accountName) {
+		System.out.println(accountName);
+		System.out.println(accountsToString());
+		// find accountName in list of user accounts
+		Account existingAccount = accounts.stream().filter(account -> account.getAccountName().equals(accountName)).findFirst().orElse(null);
```

Comment:
I don't know if it's worth making something like an `AccountManager` which you can use to check this stuff more globally, rather than tying it to a single customer. Then we can think of adding something like an `AccountFactory` to generate different types of accounts for us. Maybe we can discuss this in our next meeting.

Other than that, that's pretty cool with the `stream()` method, I had no idea Java could do this!

Reply:
Well, the idea is to check if a customer has an account, given an account name. I think it makes sense for that to be a property of the customer. Because by definition, you already have a customer object and just want to check if a property exists on that.

Example of how this could be used:
1 is this use case
Another is for moving money within your own accounts, you already have the customer object, just check if that object has the given source and destination accounts (which will be a more complex check if done globally)
A 3rd is when sending money to another customer: - Find the customer by name, check if they have the said account.

Reply:
I agree that it should have a reference locally, I just don't know if the accounts should live locally with respect to a customer. I would prefer to refactor the account so it has an ID, and that's what you use to refer to. Then you still have a list of local account ID references, just not the local account object itself. Then your use cases are still relevant, it's just decoupled.

Reply:
Okay, that makes sense for the accounts to have an ID. I've been thinking about that but only worrying about whether or not that introduces a complexity. We should discuss that with the team